### PR TITLE
feat : station command added

### PR DESCRIPTION
**Pull request by NiravThumar**
- command name : go run main.go
use : this command is root command and use for displaying welcome message and formatted output

 How to run command
>go run app.go 

rootcommand

- command name : station
use : this is command display unique station name of source station and destination station with  count in tabular format

How to run command
> go run app.go station

stationcommand

### CODE REVIEW

**Comment by NiravThumar** [comment]
- When data is empty passed No Record Found message will  show

NoRecordForStation

**Comment by pratikbgit** [comment]
@NiravThumar : I run the command : go run app.go and it will take more time but i run second time it will execute command fast.

cc: @Bhautik0110

File: data/data.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	A1            int    `csv:"a1"`
+	A2            int    `csv:"a2"`
+	A3            int    `csv:"a3"`
+	Sl            int    `csv:"sl"`
+	StationName   string `csv:"stationName"`
+	RouteNumber   int    `csv:"routeNumber"`
+	ArrivalTime   string `csv:"arrivalTime"`
+	DepartureTime string `csv:"departureTime"`
+	Distance      int    `csv:"distance"`
+}
```
Comment (Bhautik0110) [inline data/data.go:34]: make seperate file.

File: cli/main.go [hunk 2]
```diff
@@ -10,8 +13,24 @@ import (
 func Init(cfg config.AppConfig, logger *zap.Logger) error {
 	migrationCmd := GetMigrationCommandDef(cfg)
 	apiCmd := GetAPICommandDef(cfg, logger)
+	stationCmd := cmd.GetStationCmd()
+
+	rootCmd := &cobra.Command{Use: "golang-api",
+		Short: "A brief description of your application",
```
Comment (Bhautik0110) [inline cli/main.go:19]: replace with actual description.

File: cli/cmd/station.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package cmd
+
+import (
+	"sort"
+
+	"github.com/Improwised/devempire-nirav-thumar-backend/data"
+	"github.com/spf13/cobra"
+)
+
+func GetStationCmd() cobra.Command {
```
[the remark this answers is not in the document]
Comment (NiravThumar) [reply cli/cmd/station.go]: Done

File: cli/cmd/station.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+	var stationCmd = cobra.Command{
+		Use:   "station",
```
[the remark this answers is not in the document]
Comment (NiravThumar) [reply cli/cmd/station.go]: Done

File: cli/main.go (hunk 2 above)
Comment (NiravThumar) [reply cli/main.go]: Done

File: cli/cmd/station.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	var stationCmd = cobra.Command{
+		Use:   "stations",
+		Short: "Give unique station name",
+		Long:  `Command give unique station name with count`,
+		Run: func(cmd *cobra.Command, args []string) {
+			var combineStations []string // for unique station
+			trainInfoModel := models.NewTrainModal()
+			fileData := trainInfoModel.ReadFileForTrainInfo(constants.TrainInfoFile)                          // Readfile data from file
+			sourceStatioName := services.ReturnColumn(fileData, "SourceStationName")                          // getting souce data
+			destinationStationName := services.ReturnColumn(fileData, "DestinationStationName")               // gettind desttnation data
```
Comment (Bhautik0110) [inline cli/cmd/station.go:27]: should be part of model, Not service.

File: models/trainInfo.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	file, readErr := os.ReadFile(filename)
+	if readErr != nil {
+		log.Fatal(readErr)
+	}
+	unmarshalError := csvutil.Unmarshal(file, &trainInfos)
+	if unmarshalError != nil {
+		log.Fatal(unmarshalError)
+	}
+	return trainInfos
+}
```
Comment (Bhautik0110) [inline models/trainInfo.go:34]: here
`func ReadColumn`

File: models/trainInfo_test.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+		{TrainNo: 128, TrainName: "MAO-KOP SPEC", SourceStationName: "MADGOAN JN.", DestinationStationName: "CHHATRAPATI SHAHU MAHARAJ TERMINUS", Days: "Friday"},
+		{TrainNo: 290, TrainName: "PALACE ON WH", SourceStationName: "DELHI-SAFDAR JANG", DestinationStationName: "DELHI-SAFDAR JANG", Days: "Wednesday"},
+	}
+
+	actual := test.ReadFileForTrainInfo("../" + constants.TrainInfoFileTest)
+	assert.Equal(t, len(expected), len(actual), "length for trainInfo while read file expected is "+fmt.Sprintf("%d", len(expected))+" and atual getting is "+fmt.Sprintf("%d", len(actual)))
+
+	for index := range expected {
+		assert.Equal(t, expected[index], actual[index], "while read file  trainInfo expected is "+fmt.Sprintf("%v", expected[index])+" and atual getting is "+fmt.Sprintf("%v", actual[index]))
+	}
```
Comment (Bhautik0110) [inline models/trainInfo_test.go:26]: https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

File: services/returnUniqueValueOfSlice.go [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package services
+
+// Return uniqueValue
+func ReturnUniqueValueOfSlice(slice []string) []string {
```
Comment (Bhautik0110) [inline services/returnUniqueValueOfSlice.go:4]: No need.

File: models/trainInfo.go (hunk 6 above)
Comment (Bhautik0110) [reply models/trainInfo.go:34]: `func ListUniqueStationNames([]Train) []string)`

File: services/returnUniqueValueOfSlice.go (hunk 8 above)
Comment (Bhautik0110) [reply services/returnUniqueValueOfSlice.go:4]: Service is not meant for utils.

File: cli/cmd/station.go [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+			combineStations = append(append(combineStations, sourceStatioName...), destinationStationName...) // combine station
+			uniqueStation := services.ReturnUniqueValueOfSlice(combineStations)                               // get uniquevalue from combine station
```
Comment (Bhautik0110) [inline cli/cmd/station.go:29]: Make function `UniqueStationName` within `models/trainInfo.go`. Call this function.

**Review by Bhautik0110 — CHANGES_REQUESTED** [review]

File: cli/cmd/station.go (hunk 5 above)
Comment (NiravThumar) [reply cli/cmd/station.go]: move to models

File: services/returnUniqueValueOfSlice.go (hunk 8 above)
Comment (NiravThumar) [reply services/returnUniqueValueOfSlice.go]: move to utils

**Review by Bhautik0110 — APPROVED** [review]

**Pull request merged**